server/executor/runner.go

```diff
@@ -31,14 +32,18 @@ type RunResult struct {
 }
 
 type Runner interface {
-	Run(context.Context, test.Test, test.RunMetadata, environment.Environment) test.Run
+	Run(context.Context, test.Test, test.RunMetadata, environment.Environment, *testrunner.RequiredGates) test.Run
```

Comment:
I was thinking of moving both the env and test runner gates to an options struct, but I think this will be something that will change soon as part of the event-driven logic we might be applying for the next big September update 👁️